text/0001-cdk-update.md

```diff
@@ -0,0 +1,285 @@
+paths. The update command acts on one or more stacks in the application, and
+will only perform the shortcut if all changes are updateable.
+
+A stack is "updateable" if the stack is explicitly marked as updateable, and it
+contains resources supported by the update process. At this time, only Lambda
+Functions and ECS Services are supported, however more resource types will be
+added to this list in the future.
+
+At a high level, `accelerate` operates by reading the current state of the
+stack, performing a stack diff on it, and if the stack is different only in
+resources supported by update plugins, running those in lieu of a full
+deployment.
+
+#### Running Update
+
+To run a one-shot update, you can invoke the `cdk deploy --accelerated` command
+on some or all of your stacks:
+
+```
+$ cdk deploy --accelerated ApplicationStack
+Checking stack ApplicationStack for possible accelerated update
+ * LambdaFunction[AppFunction]: File asset changed
+ApplicationStack can be updated
+1/1 stacks can be updated:
+Updating LambdaFunction in ApplicationStack
+...
+Update complete!
+```
+
+If the update is to an attribute of the stack that cannot be updated, the
+command will offer to perform a full deployment:
+
+```
+$ cdk deploy --accelerated ApplicationStack
+Checking stack ApplicationStack for possible accelerated update
+ * LambdaFunction[Appfunction]: Function data changed
+ApplicationStack has changes that can not be rapidly updated.
+Perform a full deployment of ApplicationStack? [Y/n]: Y
+Deploying ApplicationStack
+...
+Done!
+```
+
+When multiple stacks are provided, update only occurs if all pending changes are
+suitable for update. Otherwise, a full deployment is done:
+
+```
+$ cdk deploy --accelerated ApplicationStack DataStack
+Checking stack ApplicationStack for possible accelerated update
+ * LambdaFunction[Appfunction]: Function data changed
+ApplicationStack has changes that can not be rapidly updated.
+Checking stack DataStack for possible accelerated update
+ * LambdaFunction[StreamFunction]: File asset changed
+DataStack can be updated
+1/2 stacks can be updated: Perform a full deployment? [Y/n]? Y
+Deploying ApplicationStack
+...
+Deploying DataStack
+...
+Done!
+```
+
+In addition to running in a one shot mode, the `cdk deploy --accelerated`
+command also has a `--watch` command line option that enable it to monitor the
+assets on disk and perform an update when they change.
+
+```
+$ cdk deploy --accelerated --watch ApplicationStack DataStack
+Checking stack ApplicationStack for possible accelerated update
+ * LambdaFunction[Appfunction]: No Changes
+ApplicationStack can *not* be updated
+Checking stack DataStack for possible accelerated update
+ * LambdaFunction[StreamFunction]: No Changes
+
+Watching stack inputs for changes:
+ * LambdaFunction[Appfunction]: <spinner>
+ * LambdaFunction[StreamFunction]: <spinner>
+
+```
+
+The watcher can only watch for file changes. For Lambda code, that means that
```

Comment:
This could catch someone by surprise. 

let's say I'm wailing away on lambda code and I add a new env variable, my IDE probably auto-saves files and when I flip back to my lambda code to continue changing code in the handler a deploy starts going out. 

Could a warning with prompt be provided in this case ?